Updates to the Cloudflare Zaraz docs (#3798)
cloudflare/cloudflare-docs: content/zaraz/reference/triggers.md

# Triggers and rules

> Triggers define the conditions under which [a tool will start an action](/zaraz/get-started/create-actions/). In most cases, your objective will be to create triggers that match specific website events that are relevant to your business. A trigger can be based on an event that happened on your website, like after clicking a button or loading specific page.

Reviewer: ```suggestion
Triggers define the conditions under which [a tool will start an action](/zaraz/get-started/create-actions/). In most cases, your objective will be to create triggers that match specific website events that are relevant to your business. A trigger can be based on an event that happened on your website, like after clicking a button or loading a specific page.
```

> These website events can be passed to Cloudflare Zaraz in a number of ways. You can use the [Track](/zaraz/web-api/track/) method of the Web API or the [data layer](/zaraz/advanced/datalayer-compatibility/). Alternatively, if you don't want to write code to track events on your website, you can configure triggers to listen to browser-side website events, with different types of rules like Click Listeners or Form Submissions.

Reviewer: ```suggestion
These website events can be passed to Cloudflare Zaraz in a number of ways. You can use the [Track](/zaraz/web-api/track/) method of the Web API or the [data layer](/zaraz/advanced/datalayer-compatibility/) call. Alternatively, if you do not want to write code to track events on your website, you can configure triggers to listen to browser-side website events, with different types of rules like Click Listeners or Form Submissions.
```

## Rule types

The exact composition of the trigger will change depending on the type of rule you choose. 

<details>
<summary>Match rule</summary>
<div>

Zaraz matches the variable you input in **Variable name** with the text under **Match string**. For a complete list of supported variables, refer to [Zaraz event and system properties](/zaraz/reference/properties-reference/).

**Trigger example: Match `zaraz.track("purchase")`**

{{<table-wrap>}}

| Rule type    | Variable name               | Match operation | Match string |
| ------------ | --------------------------- | --------------- | ------------ |
| _Match rule_ | `{{ client.__zarazTrack }}` | _Equals_        | `purchase`   |

{{</table-wrap>}}

> When matching based on a System property, you'd often want to add a second rule that matches `{{ client.__zarazTrack }}` to `Pageview`. Otherwise, your trigger will be valid for every other event happening on this page too. Refer to [**Create a trigger**](/zaraz/get-started/create-trigger/) to learn how to add more than one condition to a trigger.

Reviewer: ```suggestion
When matching based on a System property, you will often want to add a second rule that matches `{{ client.__zarazTrack }}` to `Pageview`. Otherwise, your trigger will be valid for every other event happening on this page too. Refer to [**Create a trigger**](/zaraz/get-started/create-trigger/) to learn how to add more than one condition to a trigger.
```

<summary>Timer</summary>
<div>

> Set up a timer that will fire the trigger after each **Interval**. Set your interval time in milliseconds. In **Limit** specify the number of times the interval will run, causing the trigger to fire. If you do not specify a limit, the timer will repeat for as long as the page is on.

Reviewer: ```suggestion
Set up a timer that will fire the trigger after each **Interval**. Set your interval time in milliseconds. In **Limit** specify the number of times the interval will run, causing the trigger to fire. If you do not specify a limit, the timer will repeat for as long as the page is on display.
```

**Trigger example:**

{{<table-wrap>}}

| Rule type | Interval | Limit |
| --------- | -------- | ----- |
| _Timer_   | `5000`   | `1`   |

{{</table-wrap>}}

> The above Timer will fire once, after 5 seconds. To improve the performance of a web page, you can limit a Timer trigger to a specific URL, by combining it with a Match rule. For example, to set up a timer on a specific page you can set up the following rules in a trigger:

Reviewer: ```suggestion
The above Timer will fire once, after five seconds. To improve the performance of a web page, you can limit a Timer trigger to a specific URL, by combining it with a Match rule. For example, to set up a timer on a specific page you can set up the following rules in a trigger:
```
